api.swagger.yaml

```diff
@@ -1327,6 +2242,229 @@ components:
+          type: string
+          enum:
+            - public
+            - private
+            - secret
+        members_count:
+          description: A count of members in group
+          type: integer
+          example: 150
+        author_uuid:
+          description: The uuid of the author
+          type: string
+          pattern: >-
+            ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+          example: 01234567-89ab-cdef-1234-56789abcdef0
+      type: object
+    group-search-schema:
+      allOf:
+        - $ref: '#/components/schemas/group-schema'
+    group-membership-schema:
+      allOf:
+        - $ref: '#/components/schemas/group-membership-common-schema'
+        - type: object
+          properties:
+            uuid:
+              description: The UUID of the user.
+              type: string
+              pattern: >-
+                ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+              example: 01234567-89ab-cdef-1234-56789abcdef0
+          required:
+            - uuid
+    group-membership-common-schema:
+      type: object
+      properties:
+        type:
+          description: The membership type.
+          type: string
+          enum:
+            - admin
+            - member
+      required:
+        - type
+    post-schema:
+      allOf:
+      - $ref: '#/components/schemas/post-schema-base'
+      - type: object
+        properties:
+          parent_uuid:
+            description: This is the Invotra UUID of the parent of the post.
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+          group_uuid:
+            description: This is the Invotra UUID of the group that the post belongs to.
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+            type: string
+          pin:
+            description: Whether or not this post is pinned.
+            type: boolean
+          author_uuid:
+            description: The uuid of the author
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+          likes_count:
+            description: Count of likes on the post
+            type: integer
+            example: 5
+          comment_count:
+            description: Count of comments on the post
+            type: integer
+            example: 5
+          attachments_count:
+            description: Count of attachments on the post
+            type: integer
+            example: 5
+    post-schema-create:
+      allOf:
+      - $ref: '#/components/schemas/post-schema-base'
+      - type: object
+        properties:
```

Comment:
I wonder should we also be using a uuid for the category here, rather than the name? for create and update

Reply:
Updated - for now the GET returns a string and the UUID because the endpoints are sharing the base schema.

I could change it so that get, post and put all use 3 different schemas?